_posts/2020-03-30-exploting-spatial-diversity-using-unetstack3.md

```diff
@@ -0,0 +1,98 @@
+
+Configuring and using the `Unity` agent to exploit spatial diversity is easy with just two simple steps:
+
+1. Setup the receiver nodes to cooperate.
+2. Add the `Unity` agent on the main receiver.
+
+
+#### 1. Set up for receiver nodes to cooperate (an example): 
+
+To set up the group of receivers to cooperate over a short-range network, we use [`Wormhole`](https://unetstack.net/handbook/unet-handbook_preface.html)  agent provided in the latest release UnetStack 3.1.  Transmitter node makes a transmission that is heard at all the receiver nodes. However, none of the nodes are able to successfully recover the information received, as the communication link is noisy.  In order to share the received noisy signals among the receivers, we can connect the receiver nodes using a `Wormhole`. A connection over a UDP link over a WiFi network is established by adding just a few lines of code on the receiver nodes as shown below:
+
+```groovy
+container.add 'udp', new UdpLink()            (1)
+container.add 'wormhole', new WormHole()      (2)
+wormhole.dsp = 'udp'                          (3)
+```
+Line number (1) adds a [`UdpLink`](https://unetstack.net/handbook/unet-handbook_wired_and_over_the_air_links.html) agent that implements a link protocol over UDP/IP for use over wired/wireless IP networks. The next step is to add the `Wormhole` agent and make it utilize the infrastructure provided by the `UdpLink` agent to allow the agents residing on receivers nodes to transparently talk to each other! 
+
+Line number (2) adds a `Wormhole` agent which allows the [fjåge](https://fjage.readthedocs.io/en/latest/) messages to be sent between containers over a [Unet link](https://unetstack.net/handbook/unet-handbook_introduction.html). 
+
+Line number (3) tells the `Wormhole` agent to utilize the `UdpLink` as the Unet link to share information. For more details on the `Wormhole` agent, please refer to this [link](https://unetstack.net/handbook/unet-handbook_preface.html).
+
+*Additional set up on the assisting receiver* :
+
+We may not want to share the information from all agents, instead, we are interested in messages that are published on the Physical agent's topic on the main receiver.
+```groovy
+wormhole.broadcast = [topic(phy), topic(phy, Physical.SNOOP)]
+```
+The above line of code is added only on the assisting receivers to forward only the messages received on the `Physical` agent's topic to be sent over the `Wormhole` link.
+
+#### 2. Add the `Unity` agent on the main receiver:
+
+Now that the receiver nodes are ready to cooperate, we can go ahead and add the `Unity` agent on the main receiver node. 
+
+```groovy
+container.add 'unity', new Unity()
+```
+If a user wishes to take a look at the parameters of the `Unity` agent after loading the agent, type `unity` on the node's web shell interface:
```

Comment:
Why "if"? Either ask them to do it, or not.

Reply:
Fixed.